src/fileformats/ocd_file_import.cpp

```diff
@@ -1897,55 +1897,43 @@ Symbol* OcdFileImport::getGraphicObjectSymbol(const O& ocd_object)
 		symbol.setHidden(graphic_objects_hidden);
 	};
 	
-	auto get_cached_symbol = [this](const quint64 key) {
-		return graphic_symbol_index.contains(key) ?
-		            graphic_symbol_index[key] : nullptr;
-	};
-
-	auto store_cached_symbol = [this](const quint64 key, Symbol* symbol) {
-		map->addSymbol(symbol, map->getNumSymbols());
-		graphic_symbol_index[key] = symbol;
-	};
-	
 	switch (ocd_object.type)
 	{
 	case Ocd::ObjectTypeLine:
 		{
-			const auto g_key = quint64(ocd_object.color) 
-			                   | quint64(ocd_object.line_width) << 32
-			                   | quint64(Ocd::ObjectTypeLine) << 56;
-			symbol = get_cached_symbol(g_key);
-			if (!symbol)
-			{
-				auto* line_symbol = new OcdImportedLineSymbol();
-				symbol_setup_common(*line_symbol);
-				line_symbol->color = convertColor(ocd_object.color);
-				line_symbol->line_width = convertLength(ocd_object.line_width);
-				symbol = line_symbol;
-				store_cached_symbol(g_key, symbol);
-			}
+			auto* line_symbol = new OcdImportedLineSymbol();
+			symbol_setup_common(*line_symbol);
+			line_symbol->color = convertColor(ocd_object.color);
+			line_symbol->line_width = convertLength(ocd_object.line_width);
+			symbol = line_symbol;
 		}
 		break;
 	case Ocd::ObjectTypeArea:
 		{
-			const auto g_key = quint64(ocd_object.color)
-			                   | quint64(Ocd::ObjectTypeArea) << 56;
-			symbol = get_cached_symbol(g_key);
-			if (!symbol)
-			{
-				auto* area_symbol = new OcdImportedAreaSymbol();
-				symbol_setup_common(*area_symbol);
-				area_symbol->color = convertColor(ocd_object.color);		
-				symbol = area_symbol;
-				store_cached_symbol(g_key, symbol);
-			}
+			auto* area_symbol = new OcdImportedAreaSymbol();
+			symbol_setup_common(*area_symbol);
+			area_symbol->color = convertColor(ocd_object.color);
+			symbol = area_symbol;
 		}
 		break;
 	default:
-	         addWarning(tr("Encountered an unsupported type of graphic object (%1). Skipping.").arg(ocd_object.type));
-	         break;
+		addWarning(tr("Encountered an unsupported type of graphic object (%1). Skipping.").arg(ocd_object.type));
+		break;
+	}
+	
+	if (symbol)
+	{
+		int num_symbols = map->getNumSymbols();
+		for (int i = 0; i < num_symbols; ++i)
+		{
+			if (symbol->equals(map->getSymbol(i)))
+			{
+				delete symbol;
+				return map->getSymbol(i);
+			}
+		}
```

Comment:
The hash structure was not applicable (easily) for text symbols: the hash value was previously created for line and area symbols from the type (line/area), color and width and thus fitted good into a 64 bit type. However, for text symbols there are additional properties: font size, bold/italic, font family. And for line symbols the cap and join styles had not been considered before.
There is a small runtime optimization possible that I did not apply. Instead of searching through the whole list of symbols it's sufficient to remember the index of the first auxiliary symbol and then start searching from there on.